core/src/typ.rs

```diff
@@ -128,13 +134,11 @@ pub enum EnumRowsF<ERows> {
 pub enum VarKind {
     #[default]
     Type,
-    EnumRows,
     /// `excluded` keeps track of which rows appear somewhere alongside the tail, and therefore
     /// cannot appear in the tail. For instance `forall r. { ; r } -> { x : Number ; r }` assumes
-    /// `r` does not already contain an `x` field.
-    RecordRows {
-        excluded: HashSet<Ident>,
-    },
+    EnumRows { excluded: HashSet<Ident> },
```

Comment:
Does this mean the typechecker will now refuse to instantiate `a` in say `forall a. [| 'x; a |]` with `'x`? Could this have unintended consequences somewhere?

Reply:
Indeed, the typechecker won't instantiate `a` with an enum row type that has a _conflicting declaration_ for `a`. Note that, as for record rows, it means that `forall a. [| 'x; a |]` can totally be unified with a type with a `'x` row inside. However, it will refuse to instantiate to a type with e.g. `'x Number`.

In practice, I think we had this code before the row refactoring even for enum rows, and we scrapped it because for enum rows duplication didn't matter: it was a set (without associated type), so conflict couldn't happen.

Now, I _believe_ this change might only be backward-incompatible for row types that were especially written with duplicate rows, such as `[| 'x, 'x |]`, which we should probably forbid anyway. Otherwise, unification should take care of matching existing rows. I can't think of a situation where the typechecker would generate an enum type with a duplicate row by itself.